src/middlewared/middlewared/service.py

```diff
@@ -161,13 +161,21 @@ def __init__(self, middleware):
 
 class ServiceChangeMixin:
     async def _service_change(self, service, verb):
-        enabled = (await self.middleware.call(
-            'datastore.query', 'services.services', [('srv_service', '=', service)], {'get': True}
-        ))['srv_enable']
 
-        started = await self.middleware.call(f'service.{verb}', service, {'onetime': False})
+        svc_obj = await self.middleware.call(
+            'service.query',
+            [('service', '=', service)],
+            {'get': True}
+        )
+
+        if svc_obj['state'].lower() == 'running':
+            options = {'onetime': True}
+        else:
+            options = {'onetime': False}
+
+        started = await self.middleware.call(f'service.{verb}', service, options)
 
-        if enabled and not started:
+        if svc_obj['enable'] and not started:
```

Comment:
That means if the service was set to start on boot but didnt, it will complain.

I am thinking we should only run service.{verb} if the service is running?